### PR TITLE
unittest: support ztest_new_api

### LINKED ISSUE #47420: Tests: unittest with new ZTEST API
**Describe the bug**
unittests do not use a prj.conf file, so it is not possible to set CONFIG_NEW_ZTEST_API to yes, and as a result the new API can not be used.

an example is the unittest in tests/ztest/base where the CMakeLists.txt file specifies to use the deprecated ztest API for unittesting

for the unittest for the Bluetooth LLCP we have a workaround for setting kconfig values by specifying them in an include file: tests/bluetooth/controller/mock_ctrl/include/kconfig.h
Adding CONFIG_NEW_ZTEST_API=y  to this file does not work, but gives a compile error ' fatal error: syscalls/ztest_test_new.h: No such file or directory', presumably because this file is auto-generated

**To Reproduce**
1. add the line '#define CONFIG_NEW_ZTEST_API y' to the file tests/bluetooth/controller/mock_ctrl/include/kconfig.h
2. go to the folder tests/bluetooth/controller/ctrl_api
3. run twister: 'twister -v -T .'
4. see that the build fails

**Expected behavior**

**Impact**
Can not use the new API for unit testing

**Logs and console output**

**Environment (please complete the following information):**
Running on Linux

**Additional context**

### PR DESCRIPTION
**Pull request by yperess**
Add support for the new Ztest API when using unittest.

Fixes #47420

Signed-off-by: Yuval Peress <peress@google.com>

### CODE REVIEW

**Comment by kruithofa** [comment]
Looks good, but I need a few more hours for testing this more extensively

**Review by kruithofa — APPROVED** [review]
Tested with one of the unit tests for the Bluetooth control procedures, and that worked fine.
Maybe the documentation should mention that for unit tests the line 'extra_args: CONFIG_ZTEST_NEW_API=y' needs to be added to the testcase.yaml file
When this PR is in also the unittests in tests/ztest needs updating

**Review by carlescufi — APPROVED** [review]

**Review by aaronemassey — COMMENTED** [review]
Nvm.

File: subsys/testsuite/ztest/src/ztest_new.c [hunk 1]
```diff
@@ -298,9 +310,22 @@ static int run_test(struct ztest_suite_node *suite, struct ztest_unit_test *test
 		goto out;
 	}
 
+	if (setjmp(test_skip)) {
+		ret = TC_SKIP;
+		goto out;
+	}
+
```
Comment (aaronemassey) [inline subsys/testsuite/ztest/src/ztest_new.c:317]: Don't need extra newline here.

**Review by aaronemassey — APPROVED** [review]

**Pull request merged**